src/download-plugin-manager.tsx

```diff
@@ -13,28 +13,40 @@ const {FakeEvent} = core;
 
 class DownloadPluginManager extends core.FakeEventTarget {
   private _showOverlay = false;
-  private downloadService: any;
-  private downloadMetadata: DownloadMetadata = null;
+  private downloadService: DownloadService;
+  private downloadMetadatas: DownloadMetadata[] = [null];
   private playOnClose = false;
 
-  constructor(public downloadPlugin: Download, private _config: DownloadConfig, private _logger: any) {
+  constructor(public downloadPlugin: Download, private _config: DownloadConfig, private _logger: any, private _eventManager: core.EventManager) {
     super();
-    this.downloadService = new DownloadService(this.downloadPlugin.player, this._logger);
+    this.downloadService = new DownloadService(this.downloadPlugin.player, this._logger, this._eventManager);
   }
 
-  async getDownloadMetadata(refresh = false): Promise<DownloadMetadata> {
-    if (!this.downloadMetadata || refresh) {
-      this.downloadMetadata = await this.downloadService.getDownloadMetadata(this._config);
+  isMetadataEmpty(metadata: DownloadMetadata): boolean {
```

Comment:
why does this need a separate function ? do you think there will be additional logic here in the future ?

Reply:
`isMetadataEmpty` uses it in 2 places, that's why check moved to separate method:
https://github.com/kaltura/playkit-js-downloads/pull/78/files#diff-3194df8725001bb4b9724e8da79b6078668accf54d61dfe6bdb01b9d8aa3953aR30
https://github.com/kaltura/playkit-js-downloads/pull/78/files#diff-3194df8725001bb4b9724e8da79b6078668accf54d61dfe6bdb01b9d8aa3953aR41